api/all/src/main/java/io/opentelemetry/api/trace/TraceFlags.java

```diff
@@ -45,16 +47,21 @@ static TraceFlags getSampled() {
   }
 
   /**
-   * Returns the {@link TraceFlags} converted from the given lowercase hex (base16) representation.
+   * Returns the {@link TraceFlags} converted from the given lowercase hex (base16) representation,
+   * or {@code null} if input is {@code null}, too short, or has invalid characters.
    *
    * @param src the buffer where the hex (base16) representation of the {@link TraceFlags} is.
    * @param srcOffset the offset int buffer.
    * @return the {@link TraceFlags} converted from the given lowercase hex (base16) representation.
-   * @throws NullPointerException if {@code src} is null.
-   * @throws IndexOutOfBoundsException if {@code src} is too short.
-   * @throws IllegalArgumentException if invalid characters in the {@code src}.
    */
+  @Nullable
   static TraceFlags fromHex(CharSequence src, int srcOffset) {
+    // TODO: Avoid calling `subSequence`.
+    if (src == null
+        || src.length() < srcOffset + 2
+        || !OtelEncodingUtils.isValidBase16String(src.subSequence(srcOffset, srcOffset + 2))) {
+      return null;
```

Comment:
Do you think we can return a special implementation of `TraceFlags` with a `isValid` method that returns false? It'd be nice not to return null if possible, almost everything in `api/**` tries hard to prevent NPE in calling code in the first place.

Reply:
Ah - my suggestion is the same as @trask's I think we wrote it at the same time :)